Sources/FoundationMacros/PredicateMacro.swift

```diff
@@ -393,6 +393,8 @@ extension KeyPathExprSyntax {
                 }
             case .subscript(let sub):
                 result = ExprSyntax(SubscriptCallExprSyntax(calledExpression: result, arguments: sub.arguments))
+            case .method(let method):
+                result = ExprSyntax(FunctionCallExprSyntax(calledExpression: method, arguments: method.arguments))
```

Comment:
Could you add a unit test that exercises this new code to validate that it behaves correctly? IIRC this function is uses when calling something like `$0.filter(\.foo)` in a predicate to convert the `\.foo` keypath into a valid property access that the predicate can then walk. In particular, for new functionality here we'd want to both validate that a keypath to a function works in this position for functions that `#Predicate` supports and also validate that using a keypath to an unsupported function still diagnoses the issue correctly.

Reply:
Also, for now, can you please put this case in a `#if !FOUNDATION_FRAMEWORK` block? I'll come back and remove that once the swift syntax change has landed where we need it to for the `FOUNDATION_FRAMEWORK` build, in the meantime the `FOUNDATION_FRAMEWORK` build will just use the `default` case below